src/toil/jobStores/googleJobStore.py

```diff
@@ -0,0 +1,536 @@
+import base64
+from contextlib import contextmanager
+import hashlib
+import os
+import random
+import uuid
+from StringIO import StringIO
+from bd2k.util.threading import ExceptionalThread
+import boto
+import logging
+import cPickle
+import time
+from toil.jobStores.abstractJobStore import (AbstractJobStore, NoSuchJobException,
+                                             NoSuchFileException,
+                                             ConcurrentFileModificationException)
+from toil.jobWrapper import JobWrapper
+
+log = logging.getLogger(__name__)
+
+GOOGLE_STORAGE = 'gs'
+
+
+class GoogleJobStore(AbstractJobStore):
+
+    @classmethod
+    def createJobStore(cls, jobStoreString, config=None):
+        try:
+            namePrefix, projectID = jobStoreString.split(":", 1)
+            # jobstorestring = gs:project_id:bucket
+        except ValueError:
+            # we don't have a specified projectID
+            namePrefix = jobStoreString
+            projectID = None
+        return cls(namePrefix, projectID, config)
+
+    # BOTO WILL UPDATE HEADERS WITHOUT COPYING THEM FIRST. To enforce immutability & prevent
+    # this, we use getters that return copies of our original dictionaries. reported:
+    # https://github.com/boto/boto/issues/3517
+    @property
+    def encryptedHeaders(self):
+        return self._encryptedHeaders.copy()
+
+    @encryptedHeaders.setter
+    def encryptedHeaders(self, value):
+        self._encryptedHeaders = value
+
+    @property
+    def headerValues(self):
+        return self._headerValues.copy()
+
+    @headerValues.setter
+    def headerValues(self, value):
+        self._headerValues = value
+
+    def __init__(self, namePrefix, projectID=None, config=None):
+        #  create 2 buckets
+        self.projectID = projectID
+
+        self.bucketName = namePrefix+"--toil"
```

Comment:
We could just remove the suffix if you are reasonably sure that there will only ever be one bucket in this implementation.